### ISSUE TITLE
Libraries in memory map when loaded from elf core file

### ISSUE THREAD

**Issue by ZihanYe** [angr/angr-doc#303]
Hi, I'm new to symbolic execution. I got a question:

I'm trying to load an ELF core file to angr with:
```
ap = angr.Project(self.core_file, main_opts={'backend': 'elfcore'}, force_load_libs=['/lib/x86_64-linux-gnu/libc.so.6'],rebase_granularity=0x1000)
```

and I got:

```
> ap.loader.all_objects
[<ELF Object libc-2.27.so, maps [0x0:0x3f0adf]>,
 <ELF Object ld-2.27.so, maps [0x3f1000:0x61a16f]>,
 <ExternObject Object cle##externs, maps [0x61b000:0x69b000]>,
 <ELFTLSObjectV2 Object cle##tls, maps [0x69b000:0x6b0010]>,
 <KernelObject Object cle##kernel, maps [0x6b1000:0x6b9000]>,
 <ELFCore Object core7828, maps [0x7d73aa70000:0xffffffffff600fff]>]
```
and for example the `free` symbol can be found:
```
[<Symbol "free" in libc.so.6 at 0x97950>,
 <Symbol "free" in ld-linux-x86-64.so.2 at 0x40c620>]
```
However running gdb with the same core file, I can see the libraries are mapped to different location in memory:
```
From                 To                Syms         Shared Object Library
0x00007f72d30632d0  0x00007f72d31dbc3c  Yes         /lib/x86_64-linux-gnu/libc.so.6
0x00007f72d4195f10  0x00007f72d41b4b20  Yes         /lib64/ld-linux-x86-64.so.2
```
ps: I did not disable ASLR.

It seems actual libraries used by the program are loaded inside the ```<ELFCore Object core7828, maps [0x7d73aa70000:0xffffffffff600fff]>``` object. 

Does that means I won't be able to hook libc functions like this:
```
ap.hook(addr, angr.SIM_PROCEDURES['libc']['free']())
```
as I don't know where the free function is inside the main object?

**Comment by rhelmot** [comment]
By doing the force load lib, you've loaded a _second_ copy of libc into the address space. As of now, we can't introspect into the individual binaries loaded into a corefile, so you can't do the thing you wanted to do.

Being able to do the thing you want to do would be a pretty intensive refactor, probably with https://github.com/angr/cle/issues/207 and https://github.com/angr/cle/issues/123 as dependencies. We could make it so that each library loaded in the core file would be presented as an individual object loaded into memory, and make whatever mechanism that we use to parse symbols use the in-memory data instead of the elf metadata. But that's a long ways off.

**Comment by ZihanYe** [comment]
@rhelmot  Thanks! I think I can find out address of those library functions manually in GDB and hook them. It's ok for my purpose :)

**Comment by rhelmot** [comment]
fixed by angr/cle#247
